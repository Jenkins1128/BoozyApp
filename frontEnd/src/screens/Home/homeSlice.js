import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const initialState = {
	currentState: {
		location: '',
		cuisine: null,
		priceType: 6,
		$color: 'white',
		$$color: 'white',
		$$$color: 'white',
		$$$$color: 'white',
		allCategories: '',
		alreadyFavorited: false,
		alreadyRated: false,
		restaurantsArray: [],
		initialPosition: {
			latitude: 0,
			longitude: 0
		},
		markerPosition: {
			latitude: 0,
			longitude: 0
		}
	}
};

export const getCurrentLocationDataAsync = createAsyncThunk('getCurrentLocationDataAsync/status', async (data, { rejectWithValue }) => {
	const { latitude, longitude } = data;
	try {
		const response = await axios.get('https://qvsn1ge17c.execute-api.us-east-2.amazonaws.com/latest/api/yelp/' + latitude + '/' + longitude);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const homeSlice = createSlice({
	name: 'getCurrentLocationDataAsync',
	initialState,
	reducers: {
		updateLocation: (state, { payload }) => {
			state.currentState = { ...state.currentState, location: payload.location };
		},
		updateCuisine: (state, { payload }) => {
			state.currentState = { ...state.currentState, cuisine: payload.cuisine };
		},
		updatePriceType: (state, { payload }) => {
			state.currentState = { ...state.currentState, priceType: payload.priceType };
		},
		update$color: (state, { payload }) => {
			state.currentState = { ...state.currentState, $color: payload.$color };
		},
		update$$color: (state, { payload }) => {
			state.currentState = { ...state.currentState, $$color: payload.$$color };
		},
		update$$$color: (state, { payload }) => {
			state.currentState = { ...state.currentState, $$$color: payload.$$$color };
		},
		update$$$$color: (state, { payload }) => {
			state.currentState = { ...state.currentState, $$$$color: payload.$$$$color };
		},
		updateAllCategories: (state, { payload }) => {
			state.currentState = { ...state.currentState, allCategories: payload.allCategories };
		},
		updateAlreadyFavorited: (state, { payload }) => {
			state.currentState = { ...state.currentState, alreadyFavorited: payload.alreadyFavorited };
		},
		updateAlreadyRated: (state, { payload }) => {
			state.currentState = { ...state.currentState, alreadyRated: payload.alreadyRated };
		},
		updateRestaurantsArray: (state, { payload }) => {
			state.currentState = { ...state.currentState, restaurantsArray: payload.restaurantsArray };
		},
		updateInitialPosition: (state, { payload }) => {
			state.currentState = { ...state.currentState, initialPosition: payload.initialPosition };
		},
		updateMarkerPosition: (state, { payload }) => {
			state.currentState = { ...state.currentState, markerPosition: payload.markerPosition };
		},
		reset: (state) => {
			state.currentState = {
				...state.currentState,
				location: '',
				cuisine: null,
				priceType: 6,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			};
		},
		setPrice1: (state) => {
			state.currentState = {
				...state.currentState,
				priceType: 1,
				$color: 'yellow',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'white'
			};
		},
		setPrice2: (state) => {
			state.currentState = {
				...state.currentState,
				priceType: 2,
				$color: 'white',
				$$color: 'yellow',
				$$$color: 'white',
				$$$$color: 'white'
			};
		},
		setPrice3: (state) => {
			state.currentState = {
				...state.currentState,
				priceType: 3,
				$color: 'white',
				$$color: 'white',
				$$$color: 'yellow',
				$$$$color: 'white'
			};
		},
		setPrice4: (state) => {
			state.currentState = {
				...state.currentState,
				priceType: 4,
				$color: 'white',
				$$color: 'white',
				$$$color: 'white',
				$$$$color: 'yellow'
			};
		},
		resetHomeRequestStatus: (state) => {
			state.currentState.locationDataRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder
			.addCase(getCurrentLocationDataAsync.fulfilled, (state, { payload }) => {
				state.currentState = { ...state.currentState, locationDataRequestStatus: 'fulfilled', restaurantsArray: payload };
			})
			.addCase(getCurrentLocationDataAsync.rejected, (state) => {
				state.currentState = { ...state.currentState, locationDataRequestStatus: 'rejected' };
			});
	}
});

export const {
	updateLocation,
	updateCuisine,
	updatePriceType,
	update$color,
	update$$color,
	update$$$color,
	update$$$$color,
	updateAllCategories,
	updateAlreadyFavorited,
	updateAlreadyRated,
	updateRestaurantsArray,
	updateInitialPosition,
	updateMarkerPosition,
	reset,
	setPrice1,
	setPrice2,
	setPrice3,
	setPrice4,
	resetHomeRequestStatus
} = homeSlice.actions;

export const selectHomeState = (state) => state.home.currentState;
export default homeSlice.reducer;
