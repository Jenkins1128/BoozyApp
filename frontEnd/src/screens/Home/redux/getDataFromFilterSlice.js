import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';
import { baseUrl } from '../../../helpers/constants';

const initialState = {
	currentState: {}
};

export const getDataFromFilterAsync = createAsyncThunk('getDataFromFilterAsync/status', async (data, { rejectWithValue }) => {
	const { location, priceType, cuisine } = data;
	try {
		const response = await axios.get(`${baseUrl}/yelp/${location}/${priceType}/${cuisine}`);
		return response.data;
	} catch (err) {
		return rejectWithValue(err.response.data);
	}
});

export const getDataFromFilterSlice = createSlice({
	name: 'getDataFromFilterAsync',
	initialState,
	reducers: {
		resetDataFromFilterRequestStatus: (state) => {
			state.currentState.dataFromFilterRequestStatus = 'idle';
		}
	},
	extraReducers: (builder) => {
		builder.addCase(getDataFromFilterAsync.fulfilled, (state, { payload }) => {
			Object.assign(state.currentState, {
				restaurantsArray: payload,
				dataFromFilterRequestStatus: 'fulfilled'
			});
		});
		builder.addCase(getDataFromFilterAsync.rejected, (state) => {
			state.currentState.dataFromFilterRequestStatus = 'rejected';
		});
	}
});

export const { resetDataFromFilterRequestStatus } = getDataFromFilterSlice.actions;
export const selectDataFromFilterState = (state) => state.dataFromFilter.currentState;
export default getDataFromFilterSlice.reducer;
