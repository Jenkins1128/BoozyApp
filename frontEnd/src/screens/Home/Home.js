import React, { useEffect, useRef } from 'react';
import * as Location from 'expo-location';
import { StyleSheet, View, Animated, Alert, KeyboardAvoidingView, Keyboard } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import {
	getCurrentLocationDataAsync,
	reset,
	resetHomeRequestStatus,
	selectHomeState,
	setPrice1,
	setPrice2,
	setPrice3,
	setPrice4,
	updateCuisine,
	updateInitialPosition,
	updateLocation,
	updateMarkerPosition,
	updateRestaurantsArray
} from './redux/homeSlice';
import { selectRestaurantState, viewRestaurantsAsync, resetRestaurantRequestStatus } from './redux/viewRestaurantsSlice';
import { getDataFromFilterAsync, resetDataFromFilterRequestStatus, selectDataFromFilterState } from './redux/getDataFromFilterSlice';
import BoozyMap from './BoozyMap/BoozyMap';
import Searchbar from './Searchbar/Searchbar';
import RestaurantList from './RestaurantList/RestaurantList';
import FilterView from './FilterView/FilterView';

const Home = ({ navigation }) => {
	let mapref = useRef();
	const bounceValue = useRef(new Animated.Value(1000)).current;
	const keyboardVisible = useRef(false);

	const state = useSelector(selectHomeState);
	const restaurantState = useSelector(selectRestaurantState);
	const filteredRestaurantData = useSelector(selectDataFromFilterState);
	const dispatch = useDispatch();

	useEffect(() => {
		(async () => {
			const { status } = await Location.requestForegroundPermissionsAsync();
			if (status !== 'granted') {
				setErrorMsg('Permission to access location was denied');
				return;
			}
			const location = await Location.getCurrentPositionAsync({});
			const position = { latitude: parseFloat(location.coords.latitude), longitude: parseFloat(location.coords.longitude) };
			dispatch(updateInitialPosition({ initialPosition: position }));
			dispatch(updateMarkerPosition({ markerPosition: position }));
			dispatch(getCurrentLocationDataAsync({ latitude: position.latitude, longitude: position.longitude }));
		})();
	}, []);

	//did screen focus, get last filtered data
	useEffect(() => {
		const unsubscribe = navigation.addListener('focus', () => {
			if (state.restaurantsArray.length) {
				populateRestaurants(state.restaurantsArray);
			}
		});
		return unsubscribe;
	}, [navigation]);

	//Get current location data
	useEffect(() => {
		switch (state.locationDataRequestStatus) {
			case 'idle':
				return;
			case 'rejected':
				showErrorAlert('Unable to get location. Please check your network.');
				break;
			case 'fulfilled':
				populateRestaurants(state.restaurantsArray);
				break;
			default:
				return;
		}
		dispatch(resetHomeRequestStatus());
	}, [state]);

	//Get to filtered locaiton data
	useEffect(() => {
		switch (filteredRestaurantData.dataFromFilterRequestStatus) {
			case 'idle':
				return;
			case 'rejected':
				showErrorAlert('Unable to fulfill filter. Please check your network.');
				break;
			case 'fulfilled':
				if (!filteredRestaurantData.restaurantsArray.length) {
					showErrorAlert('No places found for filter.');
				} else {
					dispatch(updateRestaurantsArray({ restaurantsArray: filteredRestaurantData.restaurantsArray }));
					populateRestaurants(filteredRestaurantData.restaurantsArray);
				}
				break;
			default:
				return;
		}
		dispatch(resetDataFromFilterRequestStatus());
	}, [filteredRestaurantData]);

	//Navigation to Restaurant Page
	useEffect(() => {
		switch (restaurantState.restaurantRequestStatus) {
			case 'idle':
				return;
			case 'rejected':
				showErrorAlert('Unable to get restaurant data. Please check your network.');
				break;
			case 'fulfilled':
				navigation.navigate('Restaurant', {
					id: restaurantState.restaurantInfo.restaurantsArray[0]['id'],
					name: restaurantState.restaurantInfo.restaurantsArray[0]['name'],
					img: restaurantState.restaurantInfo.restaurantsArray[0]['img'],
					phone: restaurantState.restaurantInfo.restaurantsArray[0]['phone'],
					rating: restaurantState.restaurantInfo.restaurantsArray[0]['rating'],
					reviews: restaurantState.restaurantInfo.restaurantsArray[0]['revCount'],
					allCategories: restaurantState.restaurantInfo.allCategories,
					alreadyFavorited: restaurantState.restaurantInfo.alreadyFavorited,
					alreadyRated: restaurantState.restaurantInfo.alreadyRated
				});
				break;
			default:
				return;
		}
		dispatch(resetRestaurantRequestStatus());
	}, [restaurantState]);

	useEffect(() => {
		const keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', () => {
			keyboardVisible.current = true; // or some other action
		});
		const keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', () => {
			keyboardVisible.current = false; // or some other action
		});

		return () => {
			keyboardDidHideListener.remove();
			keyboardDidShowListener.remove();
		};
	}, []);

	const showErrorAlert = (errorString) => {
		Alert.alert('Uh oh', errorString, [{ text: 'OK' }]);
	};

	const populateRestaurants = (restaurantsArray) => {
		if (mapref != null) {
			mapref.fitToCoordinates(restaurantsArray.map(({ lat, long }) => ({ latitude: lat, longitude: long })));
		}
	};

	const showFilterOverlay = (shown) => {
		let toValue;
		if (shown) {
			toValue = 0;
		} else {
			toValue = 1000;
		}
		Animated.spring(bounceValue, {
			toValue: toValue,
			velocity: 5,
			tension: 2,
			friction: 8,
			useNativeDriver: true
		}).start();
	};

	const getDataFromFilter = () => {
		showFilterOverlay(false);
		dispatch(getDataFromFilterAsync({ location: state.location, priceType: state.priceType, cuisine: state.cuisine }));
	};

	const viewRestaurants = (index) => {
		dispatch(viewRestaurantsAsync({ index, state }));
	};

	const setMapRef = (ref) => {
		mapref = ref;
	};

	const dismiss = () => {
		if (keyboardVisible.current) {
			Keyboard.dismiss();
		}
		if (!keyboardVisible.current) {
			showFilterOverlay(false);
		}
	};

	return (
		<KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : 'height'} style={styles.container}>
			<View style={styles.inner}>
				<BoozyMap setMapRef={setMapRef} state={state} viewRestaurants={viewRestaurants} dismiss={dismiss} />
				<Searchbar state={state} dispatch={dispatch} updateLocation={updateLocation} showFilterOverlay={() => showFilterOverlay(true)} getDataFromFilter={getDataFromFilter} dismiss={dismiss} />
				<RestaurantList state={state} viewRestaurants={viewRestaurants} dismiss={dismiss} />

				<FilterView
					bounceValue={bounceValue}
					dispatch={dispatch}
					state={state}
					reset={reset}
					updateLocation={updateLocation}
					updateCuisine={updateCuisine}
					setPrice1={setPrice1}
					setPrice2={setPrice2}
					setPrice3={setPrice3}
					setPrice4={setPrice4}
					showFilterOverlay={showFilterOverlay}
					getDataFromFilter={getDataFromFilter}
				/>
			</View>
		</KeyboardAvoidingView>
	);
};

export default Home;

const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	inner: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		justifyContent: 'center'
	}
});
