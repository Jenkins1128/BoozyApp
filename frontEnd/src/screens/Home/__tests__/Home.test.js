import React from 'react';
import { reduxRender, cleanup, render, fireEvent, waitFor } from '../../../TestHelperFiles/redux/test-utils';
import Home from '../Home';
import HomeTestFuncs from '../../../TestHelperFiles/Functions/HomeTestFuncs';
import getDataFromFilterReducer, { getDataFromFilterAsync } from '../redux/getDataFromFilterSlice';
import axios from 'axios';
import { configureStore } from '@reduxjs/toolkit';
import { baseUrl } from '../../../helpers/constants';
import homeReducer, { getCurrentLocationDataAsync } from '../redux/homeSlice';
import viewRestaurantsReducer, { viewRestaurantsAsync } from '../redux/viewRestaurantsSlice';

const createTestProps = (props) => ({
	navigation: {
		navigate: jest.fn(),
		addListener: jest.fn()
	},
	...props
});

describe('<Home />', () => {
	let props;
	let rendered;

	beforeEach(() => {
		props = createTestProps({});
		rendered = reduxRender(<Home {...props} />);
	});

	afterEach(cleanup);

	//snapshot
	it('should render correctly', () => {
		expect(rendered).toMatchSnapshot();
	});

	//content
	it('should wrap view with a flexible wrapper', () => {
		const keyboardAvoidingViewComponent = rendered.getByTestId('container');
		const given = keyboardAvoidingViewComponent.props.style;
		//padding-bottom 0 added from KeyboardAvoidingView behavior prop
		const result = [{ flex: 1 }, { paddingBottom: 0 }];
		expect(result).toMatchObject(given);
	});

	it('should render inner styles and center items', () => {
		const keyboardAvoidingViewComponent = rendered.getByTestId('inner');
		const given = keyboardAvoidingViewComponent.props.style;
		//padding-bottom 0 added from KeyboardAvoidingView behavior prop
		const result = {
			flex: 1,
			backgroundColor: '#fff',
			alignItems: 'center',
			justifyContent: 'center'
		};
		expect(result).toMatchObject(given);
	});

	//functions

	describe('showErrorAlert', () => {
		it('calls sideEffect', () => {
			jest.spyOn(HomeTestFuncs, 'showErrorAlert');
			HomeTestFuncs.showErrorAlert('Network failed...');
			expect(HomeTestFuncs.showErrorAlert).toHaveBeenCalledTimes(1);
			expect(HomeTestFuncs.showErrorAlert).toHaveBeenCalledWith('Network failed...');
		});
	});

	describe('populateRestaurants', () => {
		it('calls sideEffect', () => {
			jest.spyOn(HomeTestFuncs, 'populateRestaurants');
			HomeTestFuncs.populateRestaurants([]);
			expect(HomeTestFuncs.populateRestaurants).toHaveBeenCalledTimes(1);
			expect(HomeTestFuncs.populateRestaurants).toHaveBeenCalledWith([]);
		});
	});

	describe('showFilterOverlay', () => {
		it('calls sideEffect', () => {
			jest.spyOn(HomeTestFuncs, 'showFilterOverlay');
			HomeTestFuncs.showFilterOverlay(true);
			expect(HomeTestFuncs.showFilterOverlay).toHaveBeenCalledTimes(1);
			expect(HomeTestFuncs.showFilterOverlay).toHaveBeenCalledWith(true);
		});
	});

	describe('getDataFromFilter', () => {
		it('calls sideEffect', () => {
			jest.spyOn(HomeTestFuncs, 'getDataFromFilter');
			HomeTestFuncs.getDataFromFilter();
			expect(HomeTestFuncs.getDataFromFilter).toHaveBeenCalledTimes(1);
		});
	});

	describe('setMapRef', () => {
		it('calls sideEffect', () => {
			jest.spyOn(HomeTestFuncs, 'setMapRef');
			HomeTestFuncs.setMapRef(null);
			expect(HomeTestFuncs.setMapRef).toHaveBeenCalledTimes(1);
		});
	});

	describe('dismiss', () => {
		it('calls sideEffect', () => {
			jest.spyOn(HomeTestFuncs, 'dismiss');
			HomeTestFuncs.dismiss();
			expect(HomeTestFuncs.dismiss).toHaveBeenCalledTimes(1);
		});
	});

	//redux integration
	describe('redux', () => {
		it('should dispatch getDataFromFilterAsync and state updated fulfilled', async () => {
			const filter = {
				location: 'France',
				priceType: 2,
				cuisine: 'pizza'
			};
			const getSpy = jest.spyOn(axios, 'get').mockResolvedValueOnce([]);
			const store = configureStore({
				reducer: {
					dataFromFilter: getDataFromFilterReducer
				}
			});
			await store.dispatch(getDataFromFilterAsync(filter));
			expect(getSpy).toBeCalledWith(`${baseUrl}/yelp/${filter.location}/${filter.priceType}/${filter.cuisine}`);
			const state = store.getState();
			expect(state).toEqual({
				dataFromFilter: {
					currentState: {
						restaurantsArray: undefined,
						dataFromFilterRequestStatus: 'fulfilled'
					}
				}
			});
		});

		it('should dispatch getCurrentLocationDataAsync and state updated fulfilled', async () => {
			const coordinates = {
				latitude: 0,
				longitude: 0
			};
			const getSpy = jest.spyOn(axios, 'get').mockResolvedValueOnce([]);
			const store = configureStore({
				reducer: {
					home: homeReducer
				}
			});
			await store.dispatch(getCurrentLocationDataAsync(coordinates));
			expect(getSpy).toBeCalledWith(`${baseUrl}/yelp/${coordinates.latitude}/${coordinates.longitude}`);
			const state = store.getState();
			expect(state).toEqual({
				home: {
					currentState: {
						$$$$color: 'white',
						$$$color: 'white',
						$$color: 'white',
						$color: 'white',
						allCategories: '',
						alreadyFavorited: false,
						alreadyRated: false,
						cuisine: null,
						initialPosition: {
							latitude: 0,
							longitude: 0
						},
						location: '',
						locationDataRequestStatus: 'fulfilled',
						markerPosition: {
							latitude: 0,
							longitude: 0
						},
						priceType: 6,
						restaurantsArray: undefined
					}
				}
			});
		});

		it('should dispatch viewRestaurantsAsync and state updated', async () => {
			const restaurant = {
				index: 0,
				state: {
					index: 0,
					restaurantsArray: [
						{
							id: 0,
							name: 'Texas Roadhouse',
							categories: ['barbecue']
						}
					]
				}
			};
			let restaurantId = undefined;
			const postSpy = jest.spyOn(axios, 'post').mockResolvedValueOnce([]);
			const getSpy = jest.spyOn(axios, 'get').mockResolvedValueOnce([]);
			const store = configureStore({
				reducer: {
					restaurant: viewRestaurantsReducer
				}
			});
			await store.dispatch(viewRestaurantsAsync(restaurant));
			expect(postSpy).toBeCalledWith(`${baseUrl}/favorites`, {
				name: 'Texas Roadhouse',
				restaurantId: undefined
			});
			expect(getSpy).toBeCalledWith(`${baseUrl}/${restaurantId}/review`);
			const state = store.getState();
			expect(state).toEqual({
				restaurant: {
					currentState: {
						restaurantRequestStatus: 'rejected'
					}
				}
			});
		});
	});
});
