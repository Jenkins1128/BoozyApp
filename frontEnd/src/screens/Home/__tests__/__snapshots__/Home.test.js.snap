// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
  testID="container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="inner"
  >
    <MapView
      accessible={true}
      focusable={true}
      initialCamera={
        Object {
          "altitude": 10,
          "center": Object {
            "latitude": 0,
            "longitude": 0,
          },
          "heading": 0,
          "pitch": 0,
          "zoom": 10,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 3,
          "height": 667,
          "width": 750,
        }
      }
      testID="mapView"
    >
      <Marker
        coordinate={
          Object {
            "latitude": 0,
            "longitude": 0,
          }
        }
      />
    </MapView>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "padding": 20,
          "width": "100%",
        }
      }
      testID="searchContainer"
    >
      <View
        style={
          Object {
            "marginBottom": 10,
          }
        }
        testID="searchNearbyTextContainer"
      >
        <Text
          style={
            Object {
              "color": "#EB8873",
              "fontFamily": "Arial",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
          testID="searchNearbyText"
        >
          Search for happy hours nearby.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#F0EEEE",
            "borderRadius": 15,
            "flexDirection": "row",
            "height": 50,
            "justifyContent": "space-between",
            "paddingHorizontal": 20,
            "width": "90%",
          }
        }
        testID="searchBarBackground"
      >
        <Text />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          onChangeText={[Function]}
          onKeyPress={[Function]}
          placeholder="Enter City or Zip code"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "fontSize": 18,
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#E91E63",
            "marginRight": 20,
            "marginTop": 5,
            "opacity": 1,
          }
        }
        testID="searchButton"
      >
        <Text
          style={
            Object {
              "color": "#E91E63",
            }
          }
          testID="searchtext"
        >
          search
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 2,
          "width": "100%",
        }
      }
      testID="scrollContainer"
    >
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 30,
          "padding": 20,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateY": 1000,
            },
          ],
        }
      }
      testID="subView"
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
        testID="header"
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "flex": 1,
            }
          }
          testID="title"
        >
          <Text
            style={
              Object {
                "color": "#EB8873",
                "fontSize": 20,
                "fontWeight": "bold",
              }
            }
          >
            Filters
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
          testID="headerSettings"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 20,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="resetButton"
          >
            <Text
              style={
                Object {
                  "color": "#EB8873",
                  "fontWeight": "bold",
                }
              }
              testID="resetText"
            >
              Reset
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderRadius": 20,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="cancelButton"
          >
            <Text
              style={
                Object {
                  "color": "#EB8873",
                  "fontWeight": "bold",
                }
              }
              testID="cancelText"
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
            "justifyContent": "center",
          }
        }
        testID="filterInput"
      >
        <TextInput
          allowFontScaling={true}
          multiline={false}
          onChangeText={[Function]}
          placeholder="Enter City or Zip code"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "padding": 20,
            }
          }
          testID="locationInput"
          textAlign="center"
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          multiline={false}
          onChangeText={[Function]}
          placeholder="Enter cuisine"
          rejectResponderTermination={true}
          returnKeyType="next"
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "padding": 20,
            }
          }
          testID="cuisineInput"
          textAlign="center"
          underlineColorAndroid="transparent"
          value={null}
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 100,
              "justifyContent": "space-around",
              "width": "80%",
            }
          }
          testID="priceTypeContainer"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EB8873",
                "borderRadius": 30,
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "width": 45,
              }
            }
            testID="$Button"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              $
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EB8873",
                "borderRadius": 30,
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "width": 45,
              }
            }
            testID="$$Button"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              $$
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EB8873",
                "borderRadius": 30,
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "width": 45,
              }
            }
            testID="$$$Button"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              $$$
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EB8873",
                "borderRadius": 30,
                "height": 45,
                "justifyContent": "center",
                "opacity": 1,
                "width": 45,
              }
            }
            testID="$$$$Button"
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              $$$$
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#EB8873",
              "borderRadius": 20,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
            }
          }
          testID="applyButton"
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
            testID="applyText"
          >
            Apply
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
